salesforce/backend/query.py

```diff
@@ -483,6 +513,12 @@ def execute(self, q, args=()):
             self.results = iter([])
 
     def execute_select(self, q, args):
+        if q == "SELECT django_migrations.app, django_migrations.name FROM django_migrations":
```

Comment:
I agree with you. I wrote the same point of view in some of these issues, if I remember. Fortunately the support makemigrations is marginal and not complicated.

If you are interested about "insights": My development repository, where all "problematic" methods are implemented again, is going to implement some parts of Python DB API and to be less "hacky" and less dependent on observing new Django internals. Such code is more laborious once, but should be easier repeatedly after new Django, because it depends more on documented API. I still hesitate about backward compatibility of that direction.